prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -38,8 +38,8 @@ object FHIRBundleHelpers {
         val provenanceResource = try {
             fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
         } catch (e: NoSuchElementException) {
-            // TODO: Create Provenance here
-            throw IllegalStateException("The FHIR bundle does not contain a Provenance resource")
+            addProvenanceReference(fhirBundle)
+            fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
         }
```

Comment:
Do we need this try/catch or can we just always call `addProvenanceReference`? Could look something like this:

```
addProvenanceReference(fhirBundle)
val provenanceResource = fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
```

Or if `addProvenanceReference` is updated to return a reference to that provenance, it could be just 
```
val provenanceResource = addProvenanceReference(fhirBundle)
```

This is optional, but has two benefits:

- (minor) eliminates one of the two near-identical try/catch blocks in this path
- ensures that all provenances after this point, whether they existed already or not, have diagnostic report targets populated

Reply:
I don't know. I think I was assuming that we didn't want to change it if it already existed and was populated. I am not sure the code is set up to work with ones that are already populated, but I could be wrong. Maybe @arnejduranovic could weigh in on this?